baseapp/middleware.go

```diff
@@ -90,3 +101,34 @@ func AccessHandler(f AccessCallback) func(next http.Handler) http.Handler {
 		})
 	}
 }
+
+var DefaultOtelFilter = filters.None(
+	filters.PathPrefix("/ping"),
+	filters.PathPrefix("/api/ping"),
+	filters.PathPrefix("/health"),
+	filters.PathPrefix("/api/health"),
+	filters.PathPrefix("/debug"),
+	filters.PathPrefix("/api/debug"),
+	filters.PathPrefix("/pprof"),
+	filters.PathPrefix("/api/pprof"),
+)
+
+func NewOtelHandler() func(http.Handler) http.Handler {
+	return func(next http.Handler) http.Handler {
+		return http.HandlerFunc(func(w http.ResponseWriter, r *http.Request) {
+			requestID := ""
+			if xid, ok := hlog.IDFromRequest(r); ok {
+				requestID = xid.String()
+			}
+
+			h := othttp.NewHandler(next, r.URL.Hostname(),
+				othttp.WithMessageEvents(othttp.ReadEvents, othttp.WriteEvents),
+				othttp.WithFilter(DefaultOtelFilter),
```

Comment:
I think it could be valuable to make this easier to change. Right now, you either override the package variable or copy the whole middleware function. Maybe take the filter as an argument and then pass the default filter list when constructed in `DefaultMiddleware`?

Reply:
I was debating if it would make sense to have extra options that could be passed in. In addition to the filter, you can also change the way the span names are generated, overriding whatever is passed in. It might also be nice to change that too